pkg/provider/consulcatalog/config.go

```diff
@@ -131,6 +131,21 @@ func (c *connectCert) isReady() bool {
 	return len(c.root) > 0 && c.leaf.cert != "" && c.leaf.key != ""
 }
 
+func (c *connectCert) Equal(other *connectCert) bool {
+	if other == nil {
+		return false
+	}
```

Comment:
technically (I mean, there's nothing in Go that forbids it) the receiver (c) could also be nil, so if c and other are both nil, we should return true. I let you be the judge of whether that is relevant and you want that behaviour.

Reply:
You are right, it would certainly the behavior one expects from a function called `Equal`, also added the check for the `err`